test/test_room_api_view.py

```diff
@@ -0,0 +1,45 @@
+from typing import List
+
+import pytest
+from django.test import RequestFactory
+from rest_framework.response import Response
+
+from application.entities.room_entity import RoomEntity
+from application.repositories.interfaces import GetAllRooms
+from room.views import RoomAPIView
+
+
+@pytest.mark.django_db
+class TestRoomAPI:
+    def test_get_all_rooms_returns_empty_array_when_no_rooms_found(self, rf: RequestFactory):
+        response: Response = RoomAPIView.as_view()(rf.get('/rooms'))
+        response.render()
+
+        assert response.content == b'[]'
+
+    def test_get_all_rooms_returns_penthouse_and_basement(
```

Comment:
This test does NOT rely on a real DB - instead we inject a Mock constructed below.